shared/react-native/android/app/src/main/java/io/keybase/ossifrage/modules/ShareFiles.java

```diff
@@ -31,11 +36,28 @@ public String getName() {
     @ReactMethod
     public void share(String uriPath, String mimeType, Promise promise) {
         File file = new File(uriPath);
-        Uri fileUri = FileProvider.getUriForFile(reactContext, "io.keybase.ossifrage.fileprovider", file);
-
-        Intent intent = new Intent(Intent.ACTION_SEND)
-                .putExtra(Intent.EXTRA_STREAM, fileUri)
```

Comment:
Is this because `EXTRA_STREAM` doesn't work with text files? [This](https://stackoverflow.com/a/42571762) seems to suggest it can work with `text/*`; I didn't test it myself though.

Reply:
I'll try with `text/*`, thanks!

Reply:
![screenshot_20180424-112332](https://user-images.githubusercontent.com/59594/39206396-26cb8e3c-47b2-11e8-9643-6b019a005d31.png)

Reply:
That was me trying to share a text document.

Reply:
Oh no! Why, Android, why.

What if we gave it a e.g. `application/text` mime-type?